### PR TITLE
feat(Imperva): add waf parser

### PR DESCRIPTION
**Pull request by LeoArurault**
New ingest parser for Imperva WAF

### CODE REVIEW

File: Imperva/imperva-waf/_meta/fields.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+imperva.attack.id:
+  name: imperva.attack.id
+  type: keyword
+  description: Imperva attack id.
+
+imperva.rule.name:
+  name: imperva.rule.name
+  type: text
+  description: The threat rule name that this request triggered. For example, SQL Injection or Blocked IP (ACL).
```
Comment (otetard) [inline Imperva/imperva-waf/_meta/fields.yml:94]: Should probably be replaced by `rule.name`

File: Imperva/imperva-waf/_meta/fields.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+imperva.account.reference_id:
+  name: imperva.account.reference_id
+  type: keyword
+  description: Account level reference ID. Corresponds to the Reference ID option in the Cloud Security Console Account Settings.
+
+imperva.account.name:
+  name: imperva.account.name
+  type: keyword
+  description: The account name of the site owner.
```
Comment (otetard) [inline Imperva/imperva-waf/_meta/fields.yml:29]: Not sure that we will use that information to write detection or hunting rules. Am I wrong?

File: Imperva/imperva-waf/_meta/fields.yml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+imperva.request.headers:
+  name: imperva.request.headers
+  type: array
+  description: Request headers in JSON format, with each field represented as a name-value pair.
+
+imperva.request.x_forwarded_for:
+  name: imperva.request.x_forwarded_for
+  type: text
+  description: The X-Forwarded-For request header.
```
Comment (otetard) [inline Imperva/imperva-waf/_meta/fields.yml:44]: _May_ be replaced by `network.forwarded_ip` (not sure about this).

**Review by squioc — CHANGES_REQUESTED** [review]
LGTM unless the `imperva.rule.name` to rename with `rule.name`

**Comment by LeoArurault** [comment]
After discussions with SQU, `rule.name` will contain original string as "signature generating the event", while the parsed version will be stored as a list in `imperva.rule.name`

**Review by squioc — APPROVED** [review]

**Pull request merged**